### ISSUE TITLE
grafana-config secret is not created

### ISSUE THREAD

**Issue by Sternrassler** [prometheus-operator/kube-prometheus#1372]
**What happened?**

The grafana-Pod from the grafana deployment does not start.. All other components also work perfectly.

**How to reproduce it (as minimally and precisely as possible)**:

I have carried out the installation according to the quick start guide: ``kubectl create -f manifests/setup -f manifests``

**Environment**

* Prometheus Operator version: quay.io/prometheus-operator/prometheus-operator:v0.50.
* Kubernetes version information: v1.21.4+k3s1
* Kubernetes cluster kind: K3s
* Manifests: manifests/grafana-deployment.yaml

**Anything else we need to know?**:

When creating the Grafana pod, these errors occur:
```
Events:
  Type     Reason       Age                   From               Message
  ----     ------       ----                  ----               -------
  Normal   Scheduled    22m                   default-scheduler  Successfully assigned monitoring/grafana-8df68868b-jd8f8 to r4
  Warning  FailedMount  22m                   kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-namespace-by-workload" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  22m                   kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-pod-total" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  22m                   kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-apiserver" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  22m                   kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-node-rsrc-use" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  22m                   kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-k8s-resources-namespace" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  22m                   kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-nodes" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  22m (x2 over 22m)     kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-controller-manager" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  22m (x2 over 22m)     kubelet            MountVolume.SetUp failed for volume "grafana-dashboard-k8s-resources-cluster" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  12m (x12 over 23m)    kubelet            MountVolume.SetUp failed for volume "grafana-config" : secret "grafana-config" not found
  Warning  FailedMount  2m47s (x14 over 22m)  kubelet            (combined from similar events): Unable to attach or mount volumes: unmounted volumes=[grafana-config], unattached volumes=[grafana-dashboard-k8s-resources-namespace grafana-dashboard-k8s-resources-node grafana-config grafana-dashboard-workload-total grafana-dashboard-k8s-resources-cluster grafana-dashboard-nodes grafana-storage grafana-dashboard-proxy grafana-dashboard-controller-manager grafana-datasources grafana-dashboard-prometheus grafana-dashboard-alertmanager-overview grafana-dashboard-node-cluster-rsrc-use grafana-dashboard-node-rsrc-use kube-api-access-9lsbl grafana-dashboard-namespace-by-pod grafana-dashboard-k8s-resources-pod grafana-dashboard-scheduler grafana-dashboard-k8s-resources-workload grafana-dashboard-persistentvolumesusage grafana-dashboards grafana-dashboard-kubelet grafana-dashboard-pod-total grafana-dashboard-namespace-by-workload grafana-dashboard-prometheus-remote-write grafana-dashboard-apiserver grafana-dashboard-k8s-resources-workloads-namespace grafana-dashboard-cluster-total]: timed out waiting for the condition
```

As far as I could see, no secret was created with grafana-config in any manifest.

**Comment by xsoheilalizadeh** [comment]
I have the same issue.

**Comment by sugam72** [comment]
@Sternrassler I had the same issue. 
The problem was the compatibility mismatch of prometheus-operator and Kubernetes version.
You can check the this link - https://github.com/prometheus-operator/kube-prometheus#compatibility

In my case, the prometheus-operator version i was using is v0.80 but Kubernetes version was v1.19 which are not compatible.

**Comment by ArthurSens** [comment]
Fixed by #1373
